base/parse.jl

```diff
@@ -167,6 +167,32 @@ float(x::AbstractString) = parse(Float64,x)
 
 float{S<:AbstractString}(a::AbstractArray{S}) = map!(float, similar(a,typeof(float(0))), a)
 
+function parse{T<:AbstractFloat}(::Type{T}, s::AbstractString, base::Integer)
+    s = strip(s, ' ')
```

Comment:
As i mentioned in the commit message:

I'm not sure if it would be better to expect the user to take care of strip():
This change now allocates memory to strip the string, even when no whitespaces are present.

Parsing the following array of 500.000 strings
```julia
julia> test = [string(rand([0,1],4)...,'.', rand([0,1],5)...) for i in 1:500000]
```
with whitespace stripping
```julia
julia> @benchmark parse.(Float64, test, 2)
BenchmarkTools.Trial:
  memory estimate:  34.33 MiB
  allocs estimate:  1000041
  --------------
  minimum time:     2.188 s (0.27% GC)
  median time:      2.215 s (0.16% GC)
  mean time:        2.282 s (0.19% GC)
  maximum time:     2.443 s (0.14% GC)
  --------------
  samples:          3
  evals/sample:     1
```

and without
```julia
julia> @benchmark parse.(Float64, test, 2)
BenchmarkTools.Trial:
  memory estimate:  3.82 MiB
  allocs estimate:  41
  --------------
  minimum time:     2.242 s (0.00% GC)
  median time:      2.389 s (0.00% GC)
  mean time:        2.369 s (0.00% GC)
  maximum time:     2.477 s (0.00% GC)
  --------------
  samples:          3
  evals/sample:     1
```